AutoquoteAnnouncerSuite breaks with munit 1.1.2

https://github.com/earldouglas/sbt-war/pull/1088 broke `AutoquoteAnnouncerSuite`.


From <https://github.com/earldouglas/sectery/actions/runs/17699363303/job/50302963879#step:4:536>:

```
 ==> X sectery.usecases.announcers.AutoquoteAnnouncerSuite.Autoquote does not run in the morning  0.315s java.lang.NoClassDefFoundError: munit/internal/FutureCompat$
Caused by: java.lang.ClassNotFoundException: munit.internal.FutureCompat$
==> s sectery.usecases.announcers.AutoquoteAnnouncerSuite.Autoquote does not run in the evening skipped 0.0s
==> s sectery.usecases.announcers.AutoquoteAnnouncerSuite.Autoquote does not run on the weekend skipped 0.0s
==> s sectery.usecases.announcers.AutoquoteAnnouncerSuite.Autoquote runs during the weekday skipped 0.0s
Error:  Test suite sectery.usecases.announcers.AutoquoteAnnouncerSuite failed with java.lang.NoClassDefFoundError: munit/internal/FutureCompat$.
Error:  This may be due to the ClassLoaderLayeringStrategy (ScalaLibrary) used by your task.
Error:  To improve performance and reduce memory, sbt attempts to cache the class loaders used to load the project dependencies.
Error:  The project class files are loaded in a separate class loader that is created for each test run.
Error:  The test class loader accesses the project dependency classes using the cached project dependency classloader.
Error:  With this approach, class loading may fail under the following conditions:
Error:  
Error:   * Dependencies use reflection to access classes in your project's classpath.
Error:     Java serialization/deserialization may cause this.
Error:   * An open package is accessed across layers. If the project's classes access or extend
Error:     jvm package private classes defined in a project dependency, it may cause an IllegalAccessError
Error:     because the jvm enforces package private at the classloader level.
Error:  
Error:  These issues, along with others that were not enumerated above, may be resolved by changing the class loader layering strategy.
Error:  The Flat and ScalaLibrary strategies bundle the full project classpath in the same class loader.
Error:  To use one of these strategies, set the ClassLoaderLayeringStrategy key
Error:  in your configuration, for example:
Error:  
Error:  set use_cases / Test / classLoaderLayeringStrategy := ClassLoaderLayeringStrategy.ScalaLibrary
Error:  set use_cases / Test / classLoaderLayeringStrategy := ClassLoaderLayeringStrategy.Flat
Error:  
Error:  See ClassLoaderLayeringStrategy.scala for the full list of options.
Error:  Failed: Total 62, Failed 1, Errors 0, Passed 58, Skipped 3
Error:  Failed tests:
Error:          sectery.usecases.announcers.AutoquoteAnnouncerSuite
```

--- a/build.sbt
+++ b/build.sbt
@@ -41,7 +41,7 @@ lazy val use_cases =
       libraryDependencies += "org.ocpsoft.prettytime" % "prettytime" % "5.0.9.Final",
       libraryDependencies += "net.objecthunter" % "exp4j" % "0.4.8",
       libraryDependencies += "org.scalameta" %% "munit" % "1.1.2" % Test,
-      libraryDependencies += "org.scalameta" %% "munit-scalacheck" % "1.2.0" % Test
+      libraryDependencies += "org.scalacheck" %% "scalacheck" % "1.19.0" % Test
     )
     .dependsOn(domain, effects)
 

--- a/default.nix
+++ b/default.nix
@@ -39,7 +39,7 @@ pkgs.mkSbtDerivation {
   pname = "sectery";
   version = "0.1.0-SNAPSHOT";
 
-  depsSha256 = "sha256-ztvifV+qt15qS/JbPi8QFuC1328ettynb7URlrzeJUE=";
+  depsSha256 = "sha256-hfOETl1ACjPKMUotKVS4nnNdWqFycBYRsRLXoUOf6hU=";
 
   src = ./.;
 

--- a/modules/3-use-cases/src/test/scala/usecases/announcers/AutoquoteAnnouncerSuite.scala
+++ b/modules/3-use-cases/src/test/scala/usecases/announcers/AutoquoteAnnouncerSuite.scala
@@ -1,0 +1,140 @@
+package sectery.usecases.announcers
+
+import java.time.Instant
+import java.util.Calendar
+import java.util.TimeZone
+import org.scalacheck.Prop
+import org.scalacheck.Prop.forAll
+import org.scalacheck.Properties
+import sectery.domain.entities._
+import sectery.effects._
+import sectery.effects.id.Id
+import sectery.effects.id.given
+
+class AutoquoteAnnouncerSuite extends Properties("Autoquote announcer"):
+
+  private def getTimeMillis(
+      year: Int,
+      week: Int,
+      day: Int,
+      hour: Int,
+      minute: Int,
+      second: Int
+  ): Long =
+    new Calendar.Builder()
+      .setTimeZone(TimeZone.getTimeZone("America/Phoenix"))
+      .setWeekDate(year, week, day)
+      .setTimeOfDay(hour, minute, second)
+      .build()
+      .getTimeInMillis()
+
+  sealed class _PropertySpecifier() {
+    def update(name: String, p: => Prop): Unit =
+      val _ = property(name) = p
+  }
+
+  lazy val _property = new _PropertySpecifier()
+
+  _property("Autoquote does not run in the morning") = forAll {
+    (_year: Int, _week: Int, _day: Int, _hour: Int) =>
+
+      val timeMillis: Long =
+        getTimeMillis(
+          year = math.abs(_year % 100) + 1970,
+          week = math.abs(_week % 52) + 1, // 1-52
+          day = math.abs(_day % 7) + 1, // 1-7
+          hour = math.abs(_hour % 8), // 0-7
+          minute = 0,
+          second = 0
+        )
+
+      given now: Now[Id] with
+        override def now() = Instant.ofEpochMilli(timeMillis)
+
+      given autoquote: Autoquote[Id] with
+        override def getAutoquoteMessages() =
+          List(Tx("irc", "#foo", None, "bar"))
+
+      val obtained = new AutoquoteAnnouncer[Id].announce()
+      val expected = Nil
+
+      obtained == expected
+  }
+
+  _property("Autoquote does not run in the evening") = forAll {
+    (_year: Int, _week: Int, _day: Int, _hour: Int) =>
+
+      val timeMillis: Long =
+        getTimeMillis(
+          year = math.abs(_year % 100) + 1970,
+          week = math.abs(_week % 52) + 1, // 1-52
+          day = math.abs(_day % 7) + 1, // 1-7
+          hour = math.abs(_hour % 7) + 17, // 17-23
+          minute = 0,
+          second = 0
+        )
+
+      given now: Now[Id] with
+        override def now() = Instant.ofEpochMilli(timeMillis)
+
+      given autoquote: Autoquote[Id] with
+        override def getAutoquoteMessages() =
+          List(Tx("irc", "#foo", None, "bar"))
+
+      val obtained = new AutoquoteAnnouncer[Id].announce()
+      val expected = Nil
+
+      obtained == expected
+  }
+
+  _property("Autoquote does not run on the weekend") = forAll {
+    (_year: Int, _week: Int, _day: Int, _hour: Int) =>
+
+      val timeMillis: Long =
+        getTimeMillis(
+          year = math.abs(_year % 100) + 1970,
+          week = math.abs(_week % 52) + 1, // 1-52
+          day = if _day < 0 then 1 else 7,
+          hour = math.abs(_hour % 24), // 0-23
+          minute = 0,
+          second = 0
+        )
+
+      given now: Now[Id] with
+        override def now() = Instant.ofEpochMilli(timeMillis)
+
+      given autoquote: Autoquote[Id] with
+        override def getAutoquoteMessages() =
+          List(Tx("irc", "#foo", None, "bar"))
+
+      val obtained = new AutoquoteAnnouncer[Id].announce()
+      val expected = Nil
+
+      obtained == expected
+  }
+
+  _property("Autoquote runs during the weekday") = forAll {
+    (_year: Int, _week: Int, _day: Int, _hour: Int) =>
+
+      val timeMillis: Long =
+        getTimeMillis(
+          year = math.abs(_year % 100) + 1970,
+          week = math.abs(_week % 52) + 1, // 1-52
+          day = math.abs(_day % 5) + 2, // 2-6
+          hour = math.abs(_hour % 9) + 8, // 8-16
+          minute = 0,
+          second = 0
+        )
+
+      given now: Now[Id] with
+        override def now() = Instant.ofEpochMilli(timeMillis)
+
+      given autoquote: Autoquote[Id] with
+        override def getAutoquoteMessages() =
+          List(Tx("irc", "#foo", None, "bar"))
+
+      val obtained = new AutoquoteAnnouncer[Id].announce()
+      val expected = List(Tx("irc", "#foo", None, "bar"))
+
+      obtained == expected
+  }

